### PR TITLE
Migrating Text extraction using BERT to keras3

### PR DESCRIPTION
**Pull request by LakshmiKalaKadali**
This PR migrates the Text Extraction with BERT tutorial (Question Answering on the SQuAD dataset) from Keras 2 to Keras 3. The migration ensures the code is fully backend-agnostic, supporting JAX, PyTorch, and TensorFlow, and is optimized for stable training on specialized hardware like TPUs.
The following changes made:

1. Replaced the TensorFlow-only transformers.TFBertModel with keras_nlp.models.BertBackbone. This allows the architecture to run across all Keras 3 backends.

2. PyDataset Implementation: Replaced large in-memory NumPy arrays with keras.utils.PyDataset. This avoids Host RAM "Session Crashed" errors by streaming data in batches, which is critical for handling large BERT sequences.

3. Migrated from tf.distribute.TPUStrategy to the new Keras 3 Distribution API (DeviceMesh and DataParallel). This allows the same sharding logic to work on JAX TPUs and PyTorch Multi-GPU setups. [Colab notebook](https://colab.research.google.com/gist/LakshmiKalaKadali/7f840f6e41a7b48b077811f7a3acab66/bert-classification.ipynb#scrollTo=4JUVlZyhCK5-)

### CODE REVIEW

File: examples/nlp/text_extraction_with_bert.py [hunk 1]
```diff
@@ -279,73 +294,49 @@ def create_model():
...
         self.x_eval = x_eval
         self.y_eval = y_eval
 
     def on_epoch_end(self, epoch, logs=None):
-        pred_start, pred_end = self.model.predict(self.x_eval)
+        pred_start_logits, pred_end_logits = self.model.predict(
+            self.x_eval, batch_size=32
+        )
+        pred_start = ops.convert_to_numpy(ops.softmax(pred_start_logits))
+        pred_end = ops.convert_to_numpy(ops.softmax(pred_end_logits))
```
[the remark this answers is not in the document]
Comment (LakshmiKalaKadali) [reply examples/nlp/text_extraction_with_bert.py]: Done

File: examples/nlp/text_extraction_with_bert.py [hunk 2]
```diff
@@ -39,31 +40,41 @@
...
-# Save the slow pretrained tokenizer
-slow_tokenizer = BertTokenizer.from_pretrained("bert-base-uncased")
-save_path = "bert_base_uncased/"
-if not os.path.exists(save_path):
-    os.makedirs(save_path)
-slow_tokenizer.save_pretrained(save_path)
 
-# Load the fast tokenizer from saved file
-tokenizer = BertWordPieceTokenizer("bert_base_uncased/vocab.txt", lowercase=True)
+tokenizer_nlp = keras_hub.models.BertTokenizer.from_preset(PRESET)
```
Comment (sachinprasadhs) [inline examples/nlp/text_extraction_with_bert.py]: tokenizer_nlp --> tokenizer

File: examples/nlp/text_extraction_with_bert.py [hunk 3]
```diff
@@ -221,52 +231,58 @@ def create_inputs_targets(squad_examples):
...
 
-    start_probs = layers.Activation(keras.activations.softmax)(start_logits)
-    end_probs = layers.Activation(keras.activations.softmax)(end_logits)
+    start_logits = layers.Dense(1, use_bias=False)(embedding)
+    start_logits = layers.Reshape((max_len,), name="start_logit")(start_logits)
+    end_logits = layers.Dense(1, use_bias=False)(embedding)
+    end_logits = layers.Reshape((max_len,), name="end_logit")(end_logits)
 
     model = keras.Model(
         inputs=[input_ids, token_type_ids, attention_mask],
```
Comment (sachinprasadhs) [inline examples/nlp/text_extraction_with_bert.py]: This is defined list of inputs here, but SQuADDataset.__getitem__ has input as dict. This is ambiguous, maintain consistency.

File: examples/nlp/text_extraction_with_bert.py [hunk 4]
```diff
@@ -279,73 +295,49 @@ def create_model():
...
         count = 0
-        eval_examples_no_skip = [_ for _ in eval_squad_examples if _.skip == False]
-        for idx, (start, end) in enumerate(zip(pred_start, pred_end)):
-            squad_eg = eval_examples_no_skip[idx]
+        for idx, (start_probs, end_probs) in enumerate(zip(pred_start, pred_end)):
+            squad_eg = valid_data_no_skip[idx]
             offsets = squad_eg.context_token_to_char
-            start = np.argmax(start)
-            end = np.argmax(end)
+            start, end = np.argmax(start_probs), np.argmax(end_probs)
```
Comment (sachinprasadhs) [inline examples/nlp/text_extraction_with_bert.py]: If your ultimate aim is to make `np.argmax` of `start` and `end`.
No need to apply `ops.softmax` to `pred_start` and `pred_end`, it's a waste op.

**Review by sachinprasadhs — COMMENTED** [review]
Couple of comments to address. 
Rest all looks good. Thanks.

File: examples/nlp/text_extraction_with_bert.py (hunk 4 above)
Comment (LakshmiKalaKadali) [reply examples/nlp/text_extraction_with_bert.py]: Make sense. Addressed

File: examples/nlp/text_extraction_with_bert.py (hunk 3 above)
Comment (LakshmiKalaKadali) [reply examples/nlp/text_extraction_with_bert.py]: Made it in consistent with Dataset

File: examples/nlp/text_extraction_with_bert.py (hunk 2 above)
Comment (LakshmiKalaKadali) [reply examples/nlp/text_extraction_with_bert.py]: Done
